ancestrymanager/ancestryutil.go

```diff
@@ -0,0 +1,96 @@
+package ancestrymanager
+
+import (
+	"fmt"
+	"strings"
+
+	resources "github.com/GoogleCloudPlatform/terraform-validator/converters/google/resources"
+)
+
+// AssetParent derives a resource's parent from its ancestors.
+func AssetParent(cai *resources.Asset, ancestors []string) (string, error) {
+	if err := validateAncestors(ancestors); err != nil {
+		return "", err
+	}
+	if cai == nil {
+		return "", fmt.Errorf("asset not provided")
+	}
+	switch cai.Type {
+	case "cloudresourcemanager.googleapis.com/Folder":
+		if len(ancestors) < 2 {
+			return "", fmt.Errorf("unexpected ancestors %s", ancestors)
+		}
+		parent := ancestors[1]
+		if strings.HasPrefix(parent, "folders/") || strings.HasPrefix(parent, "organizations/") {
+			return fmt.Sprintf("//cloudresourcemanager.googleapis.com/%s", ancestors[1]), nil
+		}
+	case "cloudresourcemanager.googleapis.com/Organization":
+		return "", nil
+	case "cloudresourcemanager.googleapis.com/Project":
+		if len(ancestors) < 1 {
+			return "", fmt.Errorf("unexpected ancestors %s", ancestors)
+		}
+		if len(ancestors) > 1 {
+			return fmt.Sprintf("//cloudresourcemanager.googleapis.com/%s", ancestors[1]), nil
+		}
+		// project creation/update
+		return fmt.Sprintf("//cloudresourcemanager.googleapis.com/%s", ancestors[0]), nil
```

Comment:
Does this match the current behavior for projects? if so, we might need to make the handling of folders match this behavior for now - i.e. if the ancestry isn't known, use the folder itself.

Reply:
The current behavior just put projects/X as parent, no matter what resource it is. X is discovered by getProject, so if the resource does not have a project, then it uses the flag). 
The change is, no matter the project is newly create or not
- if the ancestor list (should at least contain the project itself) has more than itself, then it will use ancestors[1]. 
- if the ancestor list is only having the project itself (in the case project is newly created, but it does not specify folder id or org id), then it will use the project itself as parent, which is ancestors[0].